### PR TITLE
feat: Added scroll picker field

### LINKED ISSUE #16: Week and Month selector for FlutterFormInputDateTime
For a project the FlutterFormInputDateTime needs an option so you can only select weeks or months of a year instead of all the days.

### PR DESCRIPTION
**Pull request by JacquesDIconica**
Closes #16

### CODE REVIEW

File: lib/src/inputs/scroll_picker/scroll_picker_decoration.dart [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+
+class ScrollPickerDecoration {
+  ScrollPickerDecoration({
```
Comment (freekvandeven) [inline lib/src/inputs/scroll_picker/scroll_picker_decoration.dart:8]: const

File: lib/src/inputs/scroll_picker/scroll_picker_field.dart [hunk 2]
```diff
@@ -0,0 +1,119 @@
+// SPDX-FileCopyrightText: 2022 Iconica
```
Comment (freekvandeven) [inline lib/src/inputs/scroll_picker/scroll_picker_field.dart]: 2023

File: lib/src/inputs/scroll_picker/scroll_picker_field.dart [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_input_library/src/inputs/scroll_picker/scroll_picker_decoration.dart';
+
+class ScrollPicker extends StatefulWidget {
+  const ScrollPicker({
+    required this.list,
+    required this.onChanged,
+    required this.decoration,
+    this.initialIndex,
+    Key? key,
```
Comment (freekvandeven) [inline lib/src/inputs/scroll_picker/scroll_picker_field.dart:14]: super.key

File: lib/src/inputs/scroll_picker/scroll_picker_field.dart [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+      if (newIndex != selectedIndex) {
+        widget.onChanged.call(
+            (scrollController.offset / widget.decoration.itemHeight).round());
+
+        selectedIndex = newIndex;
+      }
+    });
+  }
+
```
Comment (freekvandeven) [inline lib/src/inputs/scroll_picker/scroll_picker_field.dart:67]: dispose

File: lib/src/inputs/scroll_picker/scroll_picker_decoration.dart [hunk 5]
```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2023 Iconica
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import 'package:flutter/material.dart';
+
```
Comment (Gorter-dev) [inline lib/src/inputs/scroll_picker/scroll_picker_decoration.dart:6]: Dartdoc

File: lib/src/inputs/scroll_picker/scroll_picker_field.dart [hunk 6]
```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2023 Iconica
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import 'package:flutter/material.dart';
+import 'package:flutter_input_library/src/inputs/scroll_picker/scroll_picker_decoration.dart';
+
+class ScrollPicker extends StatefulWidget {
```
Comment (Gorter-dev) [inline lib/src/inputs/scroll_picker/scroll_picker_field.dart:8]: Dartdoc

File: lib/src/inputs/scroll_picker/scroll_picker_field.dart [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  const ScrollPicker({
+    required this.list,
+    required this.onChanged,
+    required this.decoration,
+    this.initialIndex,
+    super.key,
+  });
+
+  final List<String> list;
```
Comment (Gorter-dev) [inline lib/src/inputs/scroll_picker/scroll_picker_field.dart:17]: Dartdoc

File: lib/src/inputs/scroll_picker/scroll_picker_type_extensions.dart [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    List<DateTime> result = [];
+    for (int i = start.index; i <= end.index; i++) {
+      result.add(DateTime(2024, 1, WeekDay.values[i].index + 1));
+    }
+
+    return result;
+  }
+
+  /// Creates list of Datetime with the months from start to end.
+  List<DateTime> createMonthList(Month start, Month end, int year) {
```
Comment (Gorter-dev) [inline lib/src/inputs/scroll_picker/scroll_picker_type_extensions.dart:49]: Make year optional, defaulting  to DateTime.now().year

**Review by Gorter-dev — CHANGES_REQUESTED** [review]

**Review by Gorter-dev — APPROVED** [review]

**Pull request merged**